BangazonAPI/Controllers/PaymentTypeController.cs

```diff
@@ -0,0 +1,275 @@
+
+                        };
+
+
+
+                        paymentTypes.Add(paymentType);
+                    }
+
+
+                    reader.Close();
+
+                    return Ok(paymentTypes);
+                }
+            }
+        }
+
+
+
+        //GET: Code for getting a single paymentType (active or not)
+        [HttpGet("{id}", Name = "PaymentType")]
+        public async Task<IActionResult> GetSinglePaymentType([FromRoute] int id)
+        {
+            using (SqlConnection conn = Connection)
+            {
+                conn.Open();
+                using (SqlCommand cmd = conn.CreateCommand())
+                {
+                    cmd.CommandText = $"SELECT Id, AcctNumber, [Name], CustomerId, IsActive from PaymentType WHERE Id=@id";
+                    cmd.Parameters.Add(new SqlParameter("@id", id));
+                    SqlDataReader reader = cmd.ExecuteReader();
+
+                    PaymentType paymentTypeToDisplay = null;
+
+                    while (reader.Read())
+                    {
+
+                        {
+                            paymentTypeToDisplay = new PaymentType
+                            {
+                                Id = reader.GetInt32(reader.GetOrdinal("Id")),
+                                AcctNumber = reader.GetInt32(reader.GetOrdinal("AcctNumber")),
+                                Name = reader.GetString(reader.GetOrdinal("Name")),
+                                CustomerId = reader.GetInt32(reader.GetOrdinal("CustomerId")),
+                                IsActive = reader.GetBoolean(reader.GetOrdinal("IsActive"))
+
+                            };
+                        };
+                    };
+
+
+                    reader.Close();
+
+                    return Ok(paymentTypeToDisplay);
+                }
+            }
+        }
+
+        //  POST: Code for creating a paymentType
+        [HttpPost]
+        public async Task<IActionResult> PostPaymentType([FromBody] PaymentType paymentType)
+        {
+            using (SqlConnection conn = Connection)
+            {
+                conn.Open();
+                using (SqlCommand cmd = conn.CreateCommand())
+                {
+                    
+                    cmd.CommandText = $@"INSERT INTO PaymentType (AcctNumber, [Name], CustomerId, IsActive)
+                                                    OUTPUT INSERTED.Id
+                                                    VALUES (@AcctNumber, @Name, @CustomerId, 1)";
+                    cmd.Parameters.Add(new SqlParameter("@AcctNumber", paymentType.AcctNumber));
+                    cmd.Parameters.Add(new SqlParameter("@Name", paymentType.Name));
+                    cmd.Parameters.Add(new SqlParameter("@CustomerId", paymentType.CustomerId));
+                    
+
+
+
+                    int newId = (int)cmd.ExecuteScalar();
+                    paymentType.Id = newId;
```

Comment:
I'm going to approve this because it POSTs correctly to the database, but FYI: when you make a POST request, the body of the response it sends back looks like `isActive` is false (because it's referencing the `paymentType` variable from the body of the request). But when you GET the payment type you just posted, `isActive` is indeed true-- so it works! It's just the data coming back from the POST is a little misleading. It's not a huge deal at all. File under things to refactor later post-MVP.